### PR TITLE
Testing CLI with EvalAI server on Production/Staging branch on CI/CD pipeline

### PR DESCRIPTION
**Pull request by nikochiko**
PR for a GCI task to see codebase difference. May not be merged.
Spike task: Check the possibility of testing CLI with EvalAI Server on Production/Staging Branch on CI/CD Pipeline
Report: https://github.com/nikochiko/evalai-cli/blob/test-integration-with-server/tests/integration/SPIKE.md
The build is still failing due to problems while setting up the server. Still looking into it.

To give a rough idea of the code changes:
* Added command to `git clone` the production branch of EvalAI, set up the server. (Takes 8-10 minutes)
* Change API_HOST_URL to `127.0.0.1:8000`.
* Added a temporary script to wait till server is running. The tests start after that. 
* Added a spike report.
* Added some tests

### CODE REVIEW

File: evalai/utils/config.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 AUTH_TOKEN_PATH = os.path.join(AUTH_TOKEN_DIR, AUTH_TOKEN_FILE_NAME)
 
-API_HOST_URL = os.environ.get("EVALAI_API_URL", "https://evalapi.cloudcv.org")
+API_HOST_URL = os.environ.get("EVALAI_API_URL", "http://127.0.0.1:8000")
```
Comment (pushkalkatara) [inline evalai/utils/config.py]: If we need to test for the EvalAI server on the production branch which is hosted, we need to use the host as `https://evalapi.cloudcv.org`. `http://127.0.0.1:8000` is used for the local EvalAI server.

**Comment by pushkalkatara** [comment]
Do we need the bin (images.png) files ?

File: evalai/utils/config.py (hunk 1 above)
Comment (nikochiko) [reply evalai/utils/config.py]: Yes. But making tests against the live server presented some problems as:
* An auth token would be required. Exposing an auth token would have security risks.
* The production database would have to be modified. And if we add tests for example, for making submissions, that would create a lot of duplicates on the in-use database if the test is run multiple times.

So as a walkaround, here the local EvalAI server is started from the production branch itself.
```
git clone --branch production https://github.com/Cloud-CV/EvalAI.git .travis-ci_tmp_evalai && cd .travis-ci_tmp_evalai
docker-compose up --build
```
So I tried to achieve the same thing, but with allowing more freedom over the data. 
I have included an analysis of both approaches in the report: https://github.com/nikochiko/evalai-cli/blob/test-integration-with-server/tests/integration/README.md

**Comment by nikochiko** [comment]
@pushkalkatara The images have been used for illustration in the spike report: https://github.com/nikochiko/evalai-cli/blob/test-integration-with-server/tests/integration/README.md

**Comment by nikochiko** [comment]
Closing this.
